Foxtrot/Foxtrot/Extraction/ExtractorVisitor.cs

```diff
@@ -1453,6 +1453,33 @@ public TypeNode ClosureClass
                 }
             }
 
+            public override void VisitReturn(Return Return)
+            {
+                Expression source = Return.Expression;
+                Construct sourceConstruct = source as Construct;
+
+                if (sourceConstruct != null)
+                {
+                    if (sourceConstruct.Type != null && sourceConstruct.Type.Name.Name.StartsWith(this.closureTag))
```

Comment:
Probably not in normal situation. If either of them is null it is a signal that the problem is somewhere else.